src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -162,14 +162,14 @@ public void traverse() {
         final CountingReadFilter countedFilter = makeReadFilter();
         // get the filter and transformed iterator
         final Iterator<GATKRead> readIterator = getTransformedReadStream(countedFilter).iterator();
-        // get the LIBS
-        final LocusIteratorByState libs = new LocusIteratorByState(readIterator, getDownsamplingInfo(), keepUniqueReadListInLibs(), samples, header, includeDeletions(), includeNs());
-        final Iterator<AlignmentContext> iterator = createAlignmentContextIterator(header, libs);
+
+        Iterator<AlignmentContext> iterator = AlignmentContextIteratorFactory.createAlignmentContextIterator(
```

Comment:
This will be much more readable if you construct the factory and set the values and then creates the alignment context.